src/frontend/src/binder/expr/mod.rs

```diff
@@ -118,6 +123,26 @@ impl Binder {
         .into())
     }
 
+    pub(super) fn bind_in_list(
+        &mut self,
+        expr: Expr,
+        list: Vec<Expr>,
+        _negated: bool,
+    ) -> Result<ExprImpl> {
+        // FIXME: Handle `not in`.
+        let mut bound_expr_list = vec![self.bind_expr(expr)?];
+        for elem in list {
+            bound_expr_list.push(self.bind_expr(elem)?);
+        }
+        // Only look the first two. [0] is the value, [1] is the first of target list and all
+        // subsequent elements in target list should have same type.
+        Ok(
+            FunctionCall::new_with_first_n(ExprType::In, bound_expr_list, 2)
+                .unwrap()
```

Comment:
Another variadic function is `case when`, where the func signature map lookup in `infer_type` does not work well. To be consistent with it, we can do some simple checking here and call `new_with_return_type` directly, rather than introducing another `new_xxx`. Although `new_with_return_type` will be deprecated, this makes sure they are considered together.

`unwrap()` -> `?` on type mismatch.

Note that backend currently requires the 0-th to be InputRef and remaining to be Literal of the same type. This is stricter than what PG allows.